### ISSUE TITLE
SSL Connection using CA .pem

### ISSUE THREAD

**Issue by jstnlvns** [jakob/Postico#676]
## What did you do?

SSL connections are REQUIRED to my PostgreSQL server and with other clients I am able to make this determination with the SSL settings then provide the server CA to allow me to connect.

## What did you expect to happen?

Have a way to specify the SSL Connection requirements and provide the key, cert, and/or ca file.

## What actually happened?

The connection times out:

```
could not connect to server: Operation timed out
	Is the server running on host "172.192.20.143" and accepting
	TCP/IP connections on port 5432?
```

## What software versions are you using?

Postico version:  
Version 1.5.8 (4819)

macOS version:  
Version 10.14.6 (18G95)

PostgreSQL version:  
PostgreSQL 11.5 (Ubuntu 11.5-1.pgdg18.04+1)

**Comment by jakob** [comment]
The error message indicates that the server is not reachable. Common reasons for this include:
a) a firewall prevents connections from your IP
b) the port is incorrect
c) the server is not configured to listen for connections from external interfaces

You typically fix this by either changing your firewall settings, or by using an SSH tunnel.

**Regarding SSL / TLS:**
Postico always uses SSL / TLS connections by default. If an encrypted connection is not possible, a warning is shown. By default, Postico trusts all the root CAs that ship with macOS, but you can add other CAs to your keychain, and then Postico will trust them. If a server provides a non-trusted certificate, Postico shows a prompt asking you to manually verify the certificate. Currently there is no way to provide a CA file directly for a connection in Postico.

If you want to use client certificates for connecting with Postico, make sure the files are named *.crt and *.key, then edit the connection favorite, select "Use TLS Client Certificate…" from the "Options" popup menu. Then select both the crt and key files (hold command to select multiple files in the dialog).

You can find more info in the docs here: https://eggerapps.at/postico/docs/v1.5.8/security.html

I know the current UI for TLS connections is not perfect; we're working on a new interface. If you'd like to try a beta version of Postico with the new interface, please send me an email to jakob@eggerapps.at

**Comment by jstnlvns** [comment]
Thanks @jakob for the quick response! I was able to connect using other clients using a similar configuration so I assumed it simply had to do with not being able specify the CA. After importing the CA into my keychain I was able to connect. Apologies, I hadn't made it to the Security Features section of the docs, but thanks for pointing that section out. Hopefully this will help someone else in future.
